sdk/logs/src/main/scala/org/typelevel/otel4s/sdk/logs/LogRecordRef.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2025 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.sdk.logs
+
+import cats.effect.Concurrent
+import cats.effect.Ref
+import cats.syntax.functor._
+import org.typelevel.otel4s.sdk.logs.data.LogRecordData
+
+/** See [[https://opentelemetry.io/docs/specs/otel/logs/sdk/#additional-logrecord-interfaces]]
+  */
+sealed trait LogRecordRef[F[_]] {
+
+  /** Returns an immutable instance of the [[LogRecordData]], for use in export.
+    */
+  def toLogRecordData: F[LogRecordData]
```

Comment:
Should we call it `get`? 🤔 
The current name comes from the `SpanRef`: https://github.com/typelevel/otel4s/blob/9f8ae268dc1aa5fee07dee849118b4f31413c70a/sdk/trace/src/main/scala/org/typelevel/otel4s/sdk/trace/SpanRef.scala#L58.